# Thirty lectures by UC Berkeley and East Bay Municipal Utility District survey how a billion-dollar advanced water and power utility operates today, while planning and building for the next 100 years. 
- How to incorporate innovations in materials, sensors, data networks, machine learning, and other new technologies for resilience
- How to rebuild and redesign aging dams, reservoirs, pipes and pumps
- How to build public support for investment in critical, but invisible, infrastructure
- Guide to fundamental laws, regulatory rulings, findamental research projects, and research agendas for California water utilities

```{note}
This book is interactive, which means components of a page pull in real-time data from online sources, perform data analysis, and show the results. Data is open. How code transforms data into answers is clearly visible. The code is visible in active cells, and can be run to replicate the results.
```

## What A Difference A Flood Makes: A Week in California

![Total System Storage WY 2023](Images/TotalSystemStorageWY2023Annotated.png)

```{note}
Six atmospheric rivers brought EBMUD out of drought restrictions. Three more comining in January, with Sierra snow depth doubling, raises system storage to near-capacity. Summer snow melt predictions almost double. Graph shows model predictions for alternate possibilities, from January to October 2023. Similar pattern last year had drought after March, so these predictions are conservative. Most of this rainfall cannot be stored. Some reservoirs must be drained to avoid overtopping. In future editions, today's system storage will draw directly from real-time levels at Pardee and other reservoirs.  Will add estimates of total water flow to Pacific and groundwater replenishment.
```

In [17]:
from matplotlib import rcParams, cycler
import matplotlib.pyplot as plt
import numpy as np
plt.ion()

In [18]:
# Convert EBMUD Board Meet presentation: 
# Total System Storage maximum: 770-Acre Feet 
# Convert AF to gallons, cubic feet

# Convert 770 Total Acre-Foot to gallons
Total_Acre_Feet = 770
Jan10_Acre_Feet = 640
gallons = Total_Acre_Feet  * 325853.302098
#convert gallons to cubic feet
cubic_feet = gallons / 7.48
gal = f'{round(gallons):,}'
print("Maximum Storage capacity: Acre-Feet = ", Total_Acre_Feet)
print("Jan 10,2023 Storage = ", Jan10_Acre_Feet, "Acre-Feet"," | " "Percent of Maximum = ", f'{round(Jan10_Acre_Feet / Total_Acre_Feet *100):,}',"%")
print("Maximum Storage capacity: gallons = ", gal , "gallons")
print("Maximum Storage Capacity: Cubic feet = ", f'{round(cubic_feet):,}', "Cubic Feet")


Maximum Storage capacity: Acre-Feet =  770
Jan 10,2023 Storage =  640 Acre-Feet  | Percent of Maximum =  83 %
Maximum Storage capacity: gallons =  250,907,043 gallons
Maximum Storage Capacity: Cubic feet =  33,543,722 Cubic Feet


In [19]:
%pip install folium

Note: you may need to restart the kernel to use updated packages.


In [20]:
 import folium
m = folium.Map(
    location=[37.701, -121.073],
    zoom_start=8,
    tiles='Stamen Terrain'
)

folium.Marker(
    location=[38.2574, -120.8502],
    popup='Pardee Dam',
    icon=folium.Icon(icon='cloud')
).add_to(m)

folium.Marker(
    location=[37.8922222, -122.2011111],
    popup='Orinda Water Treatment Plant',
    icon=folium.Icon(color='blue')
).add_to(m)

folium.Marker(
    location=[37.8016, -122.2734],
    popup='EBMUD Headquarters',
    icon=folium.Icon(color='red', icon='info-sign')
).add_to(m)

m

In [21]:
# HIDDEN
from datetime import date

today = date.today()

# dd/mm/YY
d1 = today.strftime("%d/%m/%Y")
print("d1 =", d1)

# Textual month, day and year	
d2 = today.strftime("%B %d, %Y %A")
print("d2 =", d2)

# mm/dd/y
d3 = today.strftime("%m/%d/%y")
print("d3 =", d3)

# Month abbreviation, day and year	
d4 = today.strftime("%b-%d-%Y")
print("d4 =", d4)

d1 = 13/01/2023
d2 = January 13, 2023 Friday
d3 = 01/13/23
d4 = Jan-13-2023


In [22]:
from datetime import datetime

# datetime object containing current date and time
now = datetime.now()
 
print("now =", now)

# dd/mm/YY H:M:S
dt_string = now.strftime("%d/%m/%Y %A %H:%M:%S")
print("date and time =", dt_string)

now = 2023-01-13 21:54:44.843953
date and time = 13/01/2023 Friday 21:54:44


In [23]:
# Import date class from datetime module
from datetime import datetime
print("now",datetime.now())
#print("time",datetime.time())
# datetime object containing current date and time
now = datetime.now()
datetime.today()

today = datetime.today()
print("Today's date is: ", today)

 
print("now2 =", now)

dt_string = now.strftime("%Y-%m-%d %A %H:%M:%S yearday=%-j Timezone= %Z ")
print("date and time =", dt_string)
# Returns the current local date

now 2023-01-13 21:54:46.785906
Today's date is:  2023-01-13 21:54:46.786405
now2 = 2023-01-13 21:54:46.786329
date and time = 2023-01-13 Friday 21:54:46 yearday=13 Timezone=  


### Here's what David Macaulay saw beneath the streets of New York. We're building the same virtual reality for the streets of Berkeley--but bringing it to life with evacuation routes, travel times, flow rates, asset inventories showing risk evaluations for each pipe element, add live sensor feeds...
![Underground](Images/DM.cover.png)
### Invisible investment

## Without underground infrastructure, surface infrastructure does not work
![Looking Up](Images/DM.look.up.png)
## Annotated lectures given in Fall of 2022 for UC Berkeley Civil Engineering 112A: Water and Wastewater Engineering. Professor Kenichi Soga, assisted by post-doctoral students Wonjun Cha, Shi-Hung Chiu.
## Thirty senior engineers from EBMUD oulined how they plan, design, build and operate a billion-dollar a year water, waste water, and power utility.
### Water utilities require major new investments to keep clean water flowing. We must rebuild and extend our critical water infrastructure now.  But this investment is not yet funded. 
-  Existing subsurface capital investments are in the trillions of dollars, possibly exceeding surface investments. But designs and budgets that worked in the past are no longer sufficient for maintaining global water supplies. The infrastructure is failing. Here's how to approach needed change.
- Past cost projections are proving to be insufficient due to accelerating challenges of climate change and aging capital plant.
- Dams, reservoirs, pipes, pumps, water collection systems and water and waste water purification **systems will require from two to five times more investment** than today's conservative financial models anticipate. [^1]
[^1]: AWWA and ASME estimates
- This site builds increasingly detailed cost projections for complete system renewal. With current technologies, rebuilding a system similar to EBMUD--1.5 million daily users, \$1 billion annual expenses, 80 MGD flow, 250 MG storage--over a 25-year period, will cost $20 billion in 2022 dollars. 
- We will outline the necessary campaign to finance this cost: by tax, by rate change, by bond issue, by cost savings, by redesign of capital plant.  Financing is possible only by building an accurate estimate of costs, to enable investors to make an accurate comparison to the costs of not rebuilding. Rebuilding at today's rate will take 140 years.[^2]
[^2]: From EBMUD Capital Investments Plan, 2022--at rate of 30 pipeline miles replaced each year, over 4,500 miles of existing distribtion pipe network
- But, if innovation could cut projected costs by a factor of two, the challenge to find necessary investment can be met. That is, save $10 billion of needed investment, and public investment is attractive. 
- Could fundamental realignment of all entities building and maintaining underground critical infrastructure change how we invest?  That is, can we merge public and investor-owned utilities? 
- As examples, examine urban infrastructure in cities that built new utility systems in the past thirty years.[^3]
[^3]: Hong Kong, Singapore, Putrajaya [Water Utilities](https://en.wikipedia.org/wiki/Water_supply_and_sanitation_in_Malaysia), [United Nations Water Joint Monitoring](https://washdata.org/), Tokyo Municipal
### Faced with massive challenges from climate change and ancient infrastructure, our utilities battle each day to protect our health. But hundreds of water main breaks a day in California show we are losing ground. What is our best investment?
- Today's **aging, failing infrastructure** requires capital investment beyond existing budgets. Even the best-managed utilities see unanticipated shortfalls due to changing weather patterns, mounting energy costs, shifting water sources, new pollution challenges, and shocks to supply chains.
  - **These investments can not be avoided or postponed.** (Think Ukraine this winter) (Insert total past capital investment, total required annual rebuilding estimates: US, EU, China, India, Sub-Saharan Africa, Japan, Korea, Brazil, Middle East)

:::{Contents}
:::

- Future pages will describe how EBMUD is rebuilding it's century-old infrastructure for the next hundred years.

#### Contents
```{contents}
```